bfsos/atomic_file.go

```diff
@@ -0,0 +1,46 @@
+package bfsos
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// atomicFile represents a file, that's written only on Close.
+type atomicFile struct {
```

Comment:
why did you decide to roll your own in the end?

Reply:
https://github.com/dchest/safefile doesn't support custom tmp dir.

So our tests fail, as it creates file in current dir.

Theoretically, I can PR it with `safefile.Create("file.bin", 0644, safefile.WithTempDir("testdata/tmp"))`, but decided it'll take longer than doing a quick internal atomicFile.

Reply:
`safefile` does just os.Rename, as you've already checked.

Leaving custom `atomicFile` implementation, at least for now, doesn't hurt anyone.